packages/tpetra/core/src/Tpetra_DirectoryImpl_def.hpp

```diff
@@ -431,10 +431,28 @@ namespace Tpetra {
 
       RCP<const Teuchos::Comm<int> > comm = map.getComm ();
       const int numProcs = comm->getSize ();
-      const global_size_t nOverP = map.getGlobalNumElements () / numProcs;
       const LO LINVALID = Teuchos::OrdinalTraits<LO>::invalid();
       LookupStatus res = AllIDsPresent;
 
+      // Find the first initialized GID for search below
+      auto it = std::find_if(
+        allMinGIDs_.begin(),
+        allMinGIDs_.end()-1,
+        [](GO const gid) { return gid != std::numeric_limits<GO>::max(); }
+      );
+
+      if (it == allMinGIDs_.end()-1) {
+        // No entries have been assigned on this process
+        res = (globalIDs.size() > 0) ? IDNotPresent : AllIDsPresent;
+        std::fill(nodeIDs.begin(), nodeIDs.end(), -1);
+        if (computeLIDs)
+          std::fill(localIDs.begin(), localIDs.end(), LINVALID);
+        return res;
+      }
+
+      const int i0 = as<GO>(std::distance(allMinGIDs_.begin(), it));
```

Comment:
Woah, you have a triple cast here -- iterator distance, to `GO`, to `int`.  You could just assign to `int` directly (with a `static_cast`), without even the `as`, since `allMinGIDs.size() == numProcs + 1` (and `MPI_Comm_size` returns an `int`).  This is harmless, but it could result in an unsigned-to-signed assignment warning, if GO is unsigned.

Reply:
ya, typo.